modules/markup/markdown/goldmark.go

```diff
@@ -184,6 +186,18 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
 			if css.ColorHandler(strings.ToLower(string(colorContent))) {
 				v.AppendChild(v, NewColorPreview(colorContent))
 			}
+		case *ast.Emphasis:
+			grandparent := n.Parent().Parent()
+			_, isInBlockquote := grandparent.(*ast.Blockquote)
+			_, blockquoteAlreadyAttentionMarked := grandparent.AttributeString(hasThisBlockquoteBeenAttentionMarked)
+			if !blockquoteAlreadyAttentionMarked && isInBlockquote {
```

Comment:
Also, what's the reason for this check?
Why would you need to check this multiple times?
As far as I understood it, you only want to check at the beginning of a quote, and you probably don't enter this block multiple times, right?
So isn't all this "once-only" checking overkill if you simply check at the beginning of the quote?

Reply:
In GitHub it's not necessarily at the start

> I'm a full sentence, highlighting the word **Note** in it.

Reply:
Oh god, I don't think that's a good idea:

> Please note that this note should not be highlighted. I'm warning you, GitHub.

Reply:
Huh? How?
Okay, we definitely don't take GitHubs approach as I don't understand it.

Reply:
I would prefer a [Set](https://github.com/go-gitea/gitea/blob/main/modules/container/set.go) here:

```go
uniqueAttentionBlockquotes := make(Set[*ast.Blockquote])
_ = ast.Walk(node, func(n ast.Node, entering bool) (ast.WalkStatus, error) {
			//...
			case *ast.Emphasis:
						bq, isBlockquote := n.Parent().Parent().(*ast.Blockquote)
						if isBlockquote && uniqueAttentionBlockquotes.Add(bq) {
									//...
						}
			//...
}
```
Is `n.Parent().Parent()` safe? There is always at minimum one parent here?

Reply:
@delvh 
`> 123 **Note** 123 **Note** 123`
> 123 **Note** 123 **Note** 123

Reply:
> Huh? How? Okay, we definitely don't take GitHubs approach as I don't understand it.

@delvh What's your desired behavior?

Reply:
Since @KN4CK3R just clarified it, I don't really have a problem anymore with it.
I forgot that I need to bolden it.

Reply:
> Is `n.Parent().Parent()` safe? There is always at minimum one parent here?

Yes, the highest an Emphasis can be is 2. (Emphasis < Paragraph < Document)

Reply:
We should most likely document that.

Reply:
> We should most likely document that.

Added a comment



> I would prefer a [Set](https://github.com/go-gitea/gitea/blob/main/modules/container/set.go) here:

Using `Set` now, it's way cleaner